packages/cdktf/lib/terraform-iterator.ts

```diff
@@ -66,6 +70,22 @@ export abstract class TerraformIterator implements ITerraformIterator {
     return new ListTerraformIterator(list);
   }
 
+  /**
+   * Creates a new iterator from a complex list. One example for this would be a list of maps.
+   * The list will be converted into a map with the mapKeyAttributeName as the key.
+   * @param list the list to iterate over
+   * @param mapKeyAttributeName the name of the attribute that should be used as the key in the map
+   *
+   * @example
+   * TODO: put AWS certificate validation example
+   */
```

Comment:
Maybe we can also link to the not-yet-existing documentation page, I think the topic is complex enough that it warrants a new section in some page